### DISCUSSION TITLE
additional JS not working

### DISCUSSION THREAD

**Discussion by AuboueixD** [requarks/wiki#5573] [Help / Questions] [answered]
Hello,

I have a wiki on which I want to add some jquery such as onclick event

I created a button with id "script" and then the jquery I added doesn't show up anything

<script>
$( document ).ready(function() {
    $( "#script" ).on( "click", function() {
        alert( "test script" );
    });  
});
</script>

**Comment by NGPixel** [comment, marked as answer]
A few things wrong with your code:

- You can't use jQuery (unless you imported that library as well)
- The DOM ready event is not the proper place to add your code because the app is not yet loaded at that point. You need to use:

```html
<script>
window.boot.register('vue', () => {
  // YOUR CODE HERE
})
</script>
```

**Reply by tim-elmer** [reply]
This got me part of the way there, but not 100%. I'm still getting null from DOM selections if I start grabbing elements in the handler. If I stick a `setTimeout(myWorkHere, 500)` in the handler for `window.boot.register('vue', ...)`, things work as expected.

Obviously a race condition is not an ideal solution.

**Reply by NGPixel** [reply]
Try `setTimeout(yourFunction)` instead (without the delay parameter), which would push the execution in the next event cycle.

**Reply by tim-elmer** [reply]
Unfortunately, that still gets me a null result (using `document.querySelector("...")`).

**Reply by NGPixel** [reply]
Looking back at where the `vue` event is called, it wasn't the best place. I added a new event in 2.5.294 called `page-ready` which gets triggered on the page mounted event instead.

So this should do the trick:
```html
<script>
window.boot.register('page-ready', () => {
  // YOUR CODE HERE
})
</script>
```

**Reply by tim-elmer** [reply]
That did the trick, thanks!

**Reply by pReya** [reply]
@NGPixel I think this should be added to the docs somewhere. Took me forever to find this hook.

**Reply by NGPixel** [reply]
@pReya [It already is...](https://docs.requarks.io/troubleshooting#how-to-execute-javascript-code-on-page-load)

**Reply by pReya** [reply]
Thanks, I was probably too blind to see it.
